addon/build-waiter.ts

```diff
@@ -1,14 +1,194 @@
+import { Primitive, TestWaiter, TestWaiterDebugInfo, WaiterName } from 'ember-test-waiters';
+
 import { DEBUG } from '@glimmer/env';
-import { TestWaiter, ITestWaiter } from 'ember-test-waiters';
-import NoopTestWaiter from './noop-test-waiter';
+import Token from './token';
+import { register } from './waiter-manager';
+
+function getNextToken(): Token {
+  return new Token();
+}
+
+/**
+ * A class providing creation, registration and async waiting functionality.
+ *
+ * @public
+ * @class TestWaiterImpl<T>
+ */
+export class TestWaiterImpl<T extends object | Primitive = Token> implements TestWaiter<T> {
+  public name: WaiterName;
+  private nextToken: () => T;
+  private isRegistered = false;
+
+  items = new Map<T, TestWaiterDebugInfo>();
+  completedOperationsForTokens = new WeakMap<Token, boolean>();
+  completedOperationsForPrimitives = new Map<Primitive, boolean>();
```

Comment:
Can you prefix all of these with `private` (in TS) and prefix with an _ to make it more clear for folks that stumble?

Reply:
For surely.

Reply:
After discussion, we are going to hold off on this for now.